src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-form/crud-form.component.ts

```diff
@@ -32,34 +36,67 @@ export class CrudFormComponent implements OnInit {
     private dataGatewayService: DataGatewayService,
     private activatedRoute: ActivatedRoute,
     private taskWrapper: TaskWrapperService,
-    private location: Location
+    private location: Location,
+    private router: Router
   ) {}
 
   ngOnInit(): void {
     this.activatedRoute.data.subscribe((data: any) => {
       this.resource = data.resource;
       this.dataGatewayService.list(`ui-${this.resource}`).subscribe((response: any) => {
-        this.title = response.forms[0].control_schema.title;
-        this.uiSchema = response.forms[0].ui_schema;
-        this.controlSchema = response.forms[0].control_schema;
+        if (this.router.url.includes('/create')) {
+          this.title = response.forms[0].control_schema.title;
+          this.uiSchema = response.forms[0].ui_schema;
+          this.controlSchema = response.forms[0].control_schema;
+        } else if (this.router.url.includes('/edit')) {
+          this.title = response.forms[1].control_schema.title;
+          this.uiSchema = response.forms[1].ui_schema;
+          this.controlSchema = response.forms[1].control_schema;
```

Comment:
With the current approach you committed it will be unbearable to extend to more forms.

We should iterate thought the `response.forms` list and find the correct form instead of deciding `0 == create` and `1 == edit`. For example here we understand the `path` is the identifier of each form, then you should look for the same path in the form.